runtime/dart_isolate.cc

```diff
@@ -280,6 +282,155 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRootIsolate(
+  // exceptions happen (log the exception, but don't terminate the app). But the
+  // default unhandled_exception_callback may terminate the app, because it is
+  // only called for the root isolate (child isolates are managed by the VM and
+  // have a different error code path). So override it to simply log the error.
+  settings.unhandled_exception_callback = [](const std::string& error,
+                                             const std::string& stack_trace) {
+    FML_LOG(ERROR) << "Unhandled exception:\n" << error << "\n" << stack_trace;
+    return true;
+  };
+
+  // The platform isolate task observer must be added on the platform thread. So
+  // schedule the add function on the platform task runner.
+  TaskObserverAdd old_task_observer_add = settings.task_observer_add;
+  settings.task_observer_add = [old_task_observer_add, platform_task_runner,
+                                weak_platform_isolate_manager](
+                                   intptr_t key, const fml::closure& callback) {
+    platform_task_runner->PostTask([old_task_observer_add,
+                                    weak_platform_isolate_manager, key,
+                                    callback]() {
+      std::shared_ptr<PlatformIsolateManager> platform_isolate_manager =
+          weak_platform_isolate_manager.lock();
+      if (platform_isolate_manager == nullptr ||
+          platform_isolate_manager->IsShutdown()) {
+        // Shutdown happened in between this task being posted, and it running.
+        // platform_isolate has already been shut down. Do nothing.
+        FML_LOG(INFO) << "Shutdown before platform isolate task observer added";
+        return;
+      }
+      old_task_observer_add(key, callback);
+    });
+  };
+
+  UIDartState::Context context(task_runners);
+  context.advisory_script_uri = isolate_group_data->GetAdvisoryScriptURI();
+  context.advisory_script_entrypoint =
+      isolate_group_data->GetAdvisoryScriptEntrypoint();
+  auto isolate_data = std::make_unique<std::shared_ptr<DartIsolate>>(
+      std::shared_ptr<DartIsolate>(
+          new DartIsolate(settings, context, platform_isolate_manager)));
+
+  IsolateMaker isolate_maker =
+      [parent_isolate](
+          std::shared_ptr<DartIsolateGroupData>* unused_isolate_group_data,
+          std::shared_ptr<DartIsolate>* isolate_data, Dart_IsolateFlags* flags,
+          char** error) {
+        return Dart_CreateIsolateInGroup(
+            /*group_member=*/parent_isolate,
+            /*name=*/"PlatformIsolate",
+            /*shutdown_callback=*/
+            reinterpret_cast<Dart_IsolateShutdownCallback>(
+                DartIsolate::SpawnIsolateShutdownCallback),
+            /*cleanup_callback=*/
+            reinterpret_cast<Dart_IsolateCleanupCallback>(
+                DartIsolateCleanupCallback),
+            /*child_isolate_data=*/isolate_data,
+            /*error=*/error);
+      };
+  Dart_Isolate platform_isolate = CreateDartIsolateGroup(
```

Comment:
I think this `CreateDartIsolateGroup` function has the wrong name, as it doesn't create an isolate group.